src/Bicep.Core/Workspaces/TemplateSpecFile.cs

```diff
@@ -5,16 +5,17 @@ namespace Bicep.Core.Workspaces
 {
     public class TemplateSpecFile : ISourceFile
     {
-        public TemplateSpecFile(Uri fileUri, string? templateSpecId, ArmTemplateFile mainTemplateFile)
+        public TemplateSpecFile(Uri fileUri, string text, string? templateSpecId, ArmTemplateFile mainTemplateFile)
         {
-            this.FileUri = fileUri;
+            this.Identifier = fileUri;
+            this.Text = text;
             this.TemplateSpecId = templateSpecId;
             this.MainTemplateFile = mainTemplateFile;
         }
 
-        public Uri FileUri { get; }
+        public Uri Identifier { get; }
 
-        public string GetOriginalSource() => MainTemplateFile.GetOriginalSource();
+        public string Text { get; }
```

Comment:
```suggestion
        public string Text => MainTemplateFile.Text;
```

Shouldn't make much of a difference, but keeping a single authortative string representation would prevent weird cases where the template spec contents differ from the spec's template's contents.

Reply:
The contents are always different, as `template` is a property within a TemplateSpec file. This change actually addresses a bug that occurred when packing a TemplateSpec file into a [SourceArchive](https://github.com/Azure/bicep/blob/cd32858dd422277e1ef6082cc9d2eba5614e5fff/src/Bicep.Core/SourceCode/SourceArchive.cs#L234) with wrong file contents. I'll add a test for it later.

Reply:
Aren't they both set to the same value on https://github.com/Azure/bicep/pull/15638/files#diff-4ed4b90021a441be8fa268e43432120a56160363d65de56b3178899036d1328dR103 ? I may have missed another call to this constructor; I left the comment thinking the contents would always be the same.